packages/react-instantsearch-hooks-dom/src/ui/InfiniteHits.tsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+
+import { cx } from './lib/cx';
+
+import type { Hit } from 'instantsearch.js';
+import type { JSXElementConstructor } from 'react';
+
+export type InfiniteHitsProps<THit> = React.ComponentProps<'div'> & {
+  hitComponent?: JSXElementConstructor<{ hit: THit }>;
+  hits: THit[];
+  isFirstPage: boolean;
+  isLastPage: boolean;
+  showPrevious?: () => void;
+  showMore: () => void;
+};
+
+function DefaultHitComponent({ hit }: { hit: Hit }) {
+  return (
+    <div style={{ wordBreak: 'break-all' }}>
+      {JSON.stringify(hit).slice(0, 100)}…
+    </div>
+  );
+}
+
+export function InfiniteHits<THit extends Hit>({
+  hitComponent: HitComponent = DefaultHitComponent,
+  hits,
+  isFirstPage,
+  isLastPage,
+  showPrevious,
+  showMore,
+  ...props
+}: InfiniteHitsProps<THit>) {
+  return (
+    <div {...props} className={cx('ais-InfiniteHits', props.className)}>
```

Comment:
Should we implement `classNames` right now or would you rather wait until https://github.com/algolia/react-instantsearch/pull/3394 is merged?

Reply:
I will merge this PR first, then update with all merged components :)